base/namedtuple.jl

```diff
@@ -0,0 +1,91 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+@generated function namedtuple(::Type{NamedTuple{names,T} where T}, args...) where names
```

Comment:
My version is useful since it causes specialization on the names tuple. But we could add

```
namedtuple(syms::NTuple{N, Symbol}, args...) where N = namedtuple(NamedTuple{syms}, args...)
```

Though also I believe @vtjnash is working on inference improvements that could make my original method unnecessary.